File: client/typescript/gamium-client-common/src/options/wait-options.ts
export interface WaitOptions {
  timeoutMs: number;
  message: string;
  intervalMs: number;
  ignoreError: boolean;
}

export const DefaultWaitOptions = (): WaitOptions => {
  return {
    timeoutMs: 5000,
    message: '',
    intervalMs: 100,
    ignoreError: true,
  };
};

File: client/python/gamium/gamium/options/wait_options.py
from typing import Optional


class WaitOptions:
    def __init__(
        self,
        timeout_ms: int = 5000,
        message: str = "",
        interval_ms: int = 100,
        ignore_error: bool = True,
    ):
        self.timeout_ms = timeout_ms
        self.message = message
        self.interval_ms = interval_ms
        self.ignore_error = ignore_error
